crates/cairo-lint-core/src/fix.rs

```diff
@@ -62,10 +62,57 @@ impl Fixer {
             first_expr.expression(db).as_syntax_node().get_text_without_trivia(db),
         )
     }
+
     pub fn fix_plugin_diagnostic(&self, db: &AnalysisDatabase, diag: &PluginDiagnostic) -> String {
         match diagnostic_kind_from_message(&diag.message) {
             CairoLintKind::DestructMatch => self.fix_destruct_match(db, diag.stable_ptr.lookup(db.upcast())),
             _ => "".to_owned(),
         }
     }
+
+    pub fn fix_double_comparison(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
+        let expr = Expr::from_syntax_node(db, node.clone());
+
+        if let Expr::Binary(binary_op) = expr {
+            let lhs = binary_op.lhs(db);
+            let rhs = binary_op.rhs(db);
+
+            if let (Expr::Binary(lhs_inner), Expr::Binary(rhs_inner)) = (&lhs, &rhs) {
+                if let (Some(lhs_var), Some(rhs_var)) =
+                    (Self::extract_variable(lhs_inner, db), Self::extract_variable(rhs_inner, db))
+                {
+                    if lhs_var == rhs_var {
+                        if matches!(
+                            (lhs_inner.op(db), rhs_inner.op(db)),
+                            (BinaryOperator::EqEq(_), BinaryOperator::LT(_))
+                                | (BinaryOperator::LT(_), BinaryOperator::EqEq(_))
+                                | (BinaryOperator::GE(_), BinaryOperator::GT(_))
+                                | (BinaryOperator::LE(_), BinaryOperator::LT(_))
+                        ) {
+                            let simplified_op = match (lhs_inner.op(db), rhs_inner.op(db)) {
+                                (BinaryOperator::EqEq(_), BinaryOperator::LT(_))
+                                | (BinaryOperator::LT(_), BinaryOperator::EqEq(_))
+                                | (BinaryOperator::LE(_), BinaryOperator::LT(_))
+                                | (BinaryOperator::GE(_), BinaryOperator::GT(_)) => "<=",
+                                _ => return node.get_text(db).to_string(),
+                            };
```

Comment:
This looks redundant you can remove the first `matches!` and directly do a match. Also i think you're missing a few patterns. For reference [here](https://github.com/rust-lang/rust-clippy/blob/5da97d006e33a4458f467e41ca43085f5e5b20ba/clippy_lints/src/operators/double_comparison.rs#L38-L50) is the `clippy` implementation